contrib_bots/lib/giphy/giphy.py

```diff
@@ -0,0 +1,128 @@
+# To use this plugin, you need to set up the Giphy API key for this bot in
+# ~/.giphy_config
+
+from __future__ import absolute_import
+from __future__ import print_function
+from six.moves.configparser import SafeConfigParser
+import requests
+import logging
+import sys
+import os
+
+def parse_giphy_config(config_file):
+    config = SafeConfigParser()
+    config.readfp(f, config_file)
```

Comment:
* `readfp` is deprecated as of Python 3.2: https://docs.python.org/3/library/configparser.html#configparser.ConfigParser.read_file.
I think you can use `read_file` here and `six` will do the correct thing but I haven't tried.